lib/fabrik/DirectorManager.js

```diff
@@ -211,6 +211,31 @@ class DirectorManager extends BaseManager {
                 previous_manifest: manifest
               }, opts.context);
               return;
+            })
+            .then(() => {
+              let updateActionContext = _.cloneDeep(actionContext);
+              _.chain(updateActionContext)
+                .set('instance_guid', this.getInstanceGuid(deploymentName))
+                .value();
+              _.chain(updateActionContext.params)
+                .set('service_id', this.service.id)
+                .set('plan_id', this.plan.id)
+                .value();
+              return this
+                .getDeploymentIps(deploymentName)
```

Comment:
add this pre-update as a supported feature. If the agents dont support this feature then the action must not be invoked. It must be implemented inline with deprovision.